plugins/ingest/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -422,6 +422,10 @@ private static void appendValues(List<Object> list, Object value) {
         throw new IllegalArgumentException("field [" + path + "] of type [" + object.getClass().getName() + "] cannot be cast to [" + clazz.getName() + "]");
     }
 
+    public String renderTemplate(TemplateService.Template template) {
```

Comment:
maybe we can internally use this method whenever we do the same rather than calling createTemplateModel and execute?